react-search-component/my-app/src/RenderedList.js

```diff
@@ -0,0 +1,37 @@
+import { useState } from 'react';
+
+export default function List({ items }: Props) {
+  const [inputValue, setInputValue] = useState('');
+
+  let newArray = items.map((text) =>
+    text.includes(inputValue) ? <li>{text}</li> : null
+  );
+
+  function handleChange(event) {
+    setInputValue(event.target.value);
+  }
+
+  let nulls = 0;
+
+  for (let i = 0; i < newArray.length; i++) {
```

Comment:
You should use `newArray.forEach`. Experienced JS programmers rarely use standard for-loops.

Even better, though, is to use `newArray.filter` to remove the null elements from the array.

Best would be to use `items.filter` above to create a new array that contains just the items that match the search text.